src/Orleans/CodeGeneration/TypeUtils.cs

```diff
@@ -1079,5 +1069,32 @@ private static string EscapeIdentifier(string identifier)
                     return identifier;
             }
         }
+
+        /// <summary>
+        /// Options for formatting type names.
+        /// </summary>
+        public class TypeFormattingOptions
+        {
+            public TypeFormattingOptions(
+                string nameSuffix = null,
+                bool includeNamespace = true,
+                bool includeGenericParameters = true,
+                bool includeTypeParameters = true,
+                char nestedClassSeparator = '.')
+            {
+
+                this.NameSuffix = nameSuffix;
+                this.IncludeNamespace = includeNamespace;
+                this.IncludeGenericTypeParameters = includeGenericParameters;
+                this.IncludeTypeParameters = includeTypeParameters;
+                this.NestedTypeSeparator = nestedClassSeparator;
+            }
+
+            public bool IncludeNamespace { get; set; }
+            public bool IncludeTypeParameters { get; set; }
+            public bool IncludeGenericTypeParameters { get; set; }
+            public char NestedTypeSeparator { get; set; }
+            public string NameSuffix { get; set; }
```

Comment:
Sure, I'll do that. I wanted to allow for initializer syntax, but I'm happy to make the setters private in lieu of a need for that and given the constructor handles all fields.